### PR TITLE
336399 site address

### PR DESCRIPTION
**Pull request by AtosSK**

### CODE REVIEW

File: src/EPR.Accreditation.Facade.Common/RESTservices/HttpSiteService.cs [hunk 1]
```diff
@@ -27,5 +26,13 @@ public class HttpSiteService : BaseHttpService, IHttpSiteService
         {
             await Put($"{accreditationExternalId}/Site", site);
         }
+
+        public async Task<Guid> CreateSite(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade.Common/RESTservices/HttpSiteService.cs:32]: Why do you need to create a site? It should exist at point of adding the address

File: src/EPR.Accreditation.Facade.Common/RESTservices/Interfaces/IHttpSiteService.cs [hunk 2]
```diff
@@ -8,7 +9,10 @@ public interface IHttpSiteService
            Guid siteExternalId);
    
         Task UpdateSite(
-            Guid accreditationExternalId,
+            Guid siteExternalId,
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade.Common/RESTservices/Interfaces/IHttpSiteService.cs]: This should not be the site id, this should be the accreditation id - there is a one to one relationship to site from accreditation, so you never need the site id (unless it is an overseas site, which i do not think this work is for)

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs [hunk 3]
```diff
@@ -90,9 +90,9 @@ public async Task<IActionResult> GetHasPermitExemption(Guid accreditationExterna
 
         [HttpGet("Site")]
         public async Task<IActionResult> GetSite(
-            Guid accreditationExternalId)
+            Guid siteExternalId)
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs]: Accreditation id

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs [hunk 4]
```diff
@@ -105,5 +105,27 @@ public async Task<IActionResult> GetHasPermitExemption(Guid accreditationExterna
 
             return Ok(taskProgress);
         }
+
+        [HttpPost("Site")]
+        [ProducesResponseType(typeof(Guid), 200)]
+        public async Task<IActionResult> CreateSite(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs:146]: No - you do not need to create a site, it already exists

File: src/EPR.Accreditation.Facade/Services/Interfaces/ISiteService.cs [hunk 5]
```diff
@@ -11,5 +12,13 @@ public interface ISiteService
         public Task UpdateExemptionReferences(
             Guid accreditationExternalId,
             IEnumerable<string> references);
+
+        public Task<Guid> CreateSite(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Services/Interfaces/ISiteService.cs:16]: Don't need create

File: src/EPR.Accreditation.Facade/Services/SiteService.cs [hunk 6]
```diff
@@ -46,5 +47,22 @@ public async Task<IEnumerable<string>> GetExemptionReferences(Guid accreditation
 
             return site;
         }
+
+        public async Task<Guid> CreateSite(
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Services/SiteService.cs:51]: Don't need create

File: src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs (hunk 3 above)
Comment (AtosSK) [reply src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs]: Done.

File: src/EPR.Accreditation.Facade.Common/RESTservices/Interfaces/IHttpSiteService.cs (hunk 2 above)
Comment (AtosSK) [reply src/EPR.Accreditation.Facade.Common/RESTservices/Interfaces/IHttpSiteService.cs]: Done.

**Review by NikPeters-Atos — APPROVED** [review]

**Review by NikPeters-Atos — APPROVED** [review]

**Pull request merged**